docs/configdoc.md

```diff
@@ -0,0 +1,103 @@
+## [l3afd]
+| FieldName     | Example       | Description     | Required        |
+| ------------- | ------------- | --------------- | --------------- |
+|pid-file| `"./l3afd.pid"`  | The pin Path for l3afd.pid file which contains process id of L3afd | Yes |
+|datacenter| `"dummy"` | Name of Datacenter| Yes |
+|bpf-dir| `"/dev/shm"` | Absolute Path of eBPF Programs (where they are extracted) | Yes |
+|bpf-log-dir|`""`      | Absolute Path, where we store logs of eBPF Programs. We are not storing logs of l3afd actions on eBPF Programs| No |
+|kernel-major-version|`"4"`|Major version of the kernel| Only on linux |
+|kernel-minor-version|`"15"`|Minor version of the kernel (Ex 4.15)| Only on linux |
+|shutdown-timeout|`"1s"`|It is the amount of time allowed for L3afd graceful stop ( stopping l3afd).After shutdown-timeout L3afd itself exit| No |
+|http-client-timeout|`"10s"`|Maximum amount of time allowed to get HTTP response headers when fetching a program from a repository| No |
+|max-nf-restart-count|`"3"`|Maximum number of try's to restart eBPF programs if they are not running| No |
+|bpf-chaining-enabled|`"true"`|Boolean to set bpf-chaining. For more info about bpf chaining check [L3AF_KFaaS.pdf](https://github.com/l3af-project/l3af-arch/blob/main/L3AF_KFaaS.pdf)| Yes |
+|swagger-api-enabled|`"false"`|Whether the swagger API is enabled or not.  For more info see [swagger.md](https://github.com/l3af-project/l3afd/blob/main/docs/swagger.md)| No |
+|environment|`"PROD"`|If set to anything other than "PROD", mTLS security will not be checked| Yes |
+|BpfMapDefaultPath|`"/sys/fs/bpf"`|The base pin path for eBPF maps| Yes |
+
+## [kf-repo]
+| FieldName     | Example       | Description     | Required        |
+| ------------- | ------------- | --------------- | --------------- |
+|url| `"http://localhost:8000/"`|Default repository from which to download eBPF programs| No |
+
+## [web]
+| FieldName     | Example       | Description     |  Required        |
+| ------------- | ------------- | --------------- |  --------------- |
+|metrics-addr|`"0.0.0.0:8898"`|Prometheus endpoint for pulling/scraping the metrics.  For more info about Prometheus see [prometheus.io](https://prometheus.io/) | Yes |
+|kf-poll-interval|`"30s"`|Periodic interval at which to scrape metrics using Prometheus| Yes |
+|n-metric-samples|`"20"`|Number of Metric Samples| Yes |
+
+
+## [xdp-root-program]
+| FieldName     | Example       | Description     | Required        |
+| ------------- | ------------- | --------------- | --------------- |
+|name|`"xdp-root"`|Name of xdp-root program| Yes |
+|artifact|`"xdp-root.tar.gz"`|Archive of xdp-root program. Only tar.gz and .zip formats are supported| Yes |
+|ingress-map-name|`"root_array"`|Ingress map name of xdp-root program| Yes |
+|command|`"xdp_root"`|Command to run xdp-root program| Yes |
+|version|`"1.01"`|Version of xdp-root program| Yes |
+|user-program-daemon|`"false"`|Boolean to check xdp-root is user-program daemon or not| Yes |
```

Comment:
Explain what this means in more detail.  I think it means if false, the program will exit when started, and if true, then it will run until stopped by l3afd.  Correct?
```suggestion
|user-program-daemon|`"false"`|Set to true it requires l3afd to stop the application (via SIGTERM on Linux or SIGKILL on Windows)| Yes |
```